api/src/services/items.ts

```diff
@@ -706,18 +752,33 @@ export class ItemsService<Item extends AnyItem = AnyItem> implements AbstractSer
 
 			const childrenRevisions = [...revisionsM2O, ...revisionsA2O];
 
+			let userIntegrityCheckFlags =
+				opts.userIntegrityCheckFlags ??
+				UserIntegrityCheckFlag.None | userIntegrityCheckFlagsM2O | userIntegrityCheckFlagsA2O;
+
 			nestedActionEvents.push(...nestedActionEventsM2O);
 			nestedActionEvents.push(...nestedActionEventsA2O);
 
 			for (const key of keys) {
-				const { revisions, nestedActionEvents: nestedActionEventsO2M } = await payloadService.processO2M(
-					payloadWithA2O,
-					key,
-					opts,
-				);
+				const {
+					revisions,
+					nestedActionEvents: nestedActionEventsO2M,
+					userIntegrityCheckFlags: userIntegrityCheckFlagsO2M,
+				} = await payloadService.processO2M(payloadWithA2O, key, opts);
 
 				childrenRevisions.push(...revisions);
 				nestedActionEvents.push(...nestedActionEventsO2M);
+				userIntegrityCheckFlags ||= userIntegrityCheckFlagsO2M;
```

Comment:
Fix this @hanneskuettner! Now! Needs to be `|=`

Reply:
![gogo](https://github.com/directus/directus/assets/5363448/91a083f1-1703-4ce5-8655-947ccfff35ca)